pkg/util/unstructured.go

```diff
@@ -20,14 +20,27 @@ import (
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	listVersion = "v1"
+	listKind    = "List"
 )
 
 // ToUnstructuredList is to convert an object to unstructured.UnstructuredList.
 // If the object is not a list type, it will convert to a single item UnstructuredList.
 func ToUnstructuredList(obj runtime.Object) (*unstructured.UnstructuredList, error) {
 	unstructuredList := &unstructured.UnstructuredList{}
+	unstructuredList.SetGroupVersionKind(schema.FromAPIVersionAndKind(listVersion, listKind))
+	unstructuredList.GetObjectKind().SetGroupVersionKind(schema.FromAPIVersionAndKind(listVersion, listKind))
+
 	if meta.IsListType(obj) {
-		unstructuredList.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		if !obj.GetObjectKind().GroupVersionKind().Empty() {
+			unstructuredList.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+			unstructuredList.GetObjectKind().SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
```

Comment:
Just revisiting this PR, but do these two lines do not perform the very same operation ? Looking as the code isn't 

```
func (u *UnstructuredList) GetObjectKind() schema.ObjectKind { return u }
```

just an identity operation ?

Also I just debugged the method, and if you enter it with an `runtime.Object` that has GVK set (like for all objects that we return from the client in out client API where we explicitly set the GVK coordinates), then the conversion to unstructured works as expected so that it can be used by the printer package. Only when those GVK in the incoming object are _not set_, then the original code has an issue.